src/Contao/View/Contao2BackendView/EditMask.php

```diff
@@ -642,16 +642,21 @@ protected function getHeadline()
      */
     protected function doPersist()
     {
-        $environment   = $this->getEnvironment();
-        $dataProvider  = $environment->getDataProvider($this->model->getProviderName());
-        $inputProvider = $environment->getInputProvider();
+        $environment     = $this->getEnvironment();
+        $dataProvider    = $environment->getDataProvider($this->model->getProviderName());
+        $inputProvider   = $environment->getInputProvider();
+        $editInformation = System::getContainer()->get('cca.dc-general.edit-information');
```

Comment:
Please use dependency injection

Reply:
@dmolineus I am unsure how we should handle this.
I mean, the current code is already using the container directly, however only in a narrow context.
Now we would end up using it always for every save operation.

I tend towards using DI here, but we would still need the fallback non the less.

@zonky2 If you want to change it in 2.3, why not already now when you are already altering the code. Where is the reasoning to revisit the exact same code twice?

Reply:
On the other hand, this stateful service still feels so wrong to me.

Reply:
fixed